api/api_test.go

```diff
@@ -0,0 +1,44 @@
+package api
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+// TestCleanCloseHandler checks that if the handler keeps writing
+// after cleanCloseHandler cancels it, no race condition happens.
+// See https://github.com/NebulousLabs/Sia/issues/2385.
+func TestCleanCloseHandler(t *testing.T) {
+	t.Parallel()
```

Comment:
All tests that sleep, and all tests that touch disk in any way at all need to be skipped `if testing.Short()`. General rule of thumb for short tests is that the entire suite of short tests for a package should complete in under 0.5 seconds.